transport/pool.go

```diff
@@ -23,15 +24,19 @@ type ConnPool struct {
 	connObs      ConnObserver
 }
 
-func NewConnPool(host string, cfg ConnConfig) (*ConnPool, error) {
+func NewConnPool(ctx context.Context, host string, cfg ConnConfig) (*ConnPool, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
```

Comment:
no point of checking it here, you should check it just before/during/after blocking operations